### PR TITLE
Use framework::Client everywhere.

### PR DESCRIPTION
**Pull request by cschuet**
This switches all gRPC method invocation to using framework::Client. After this change we can remove the gRPC proto compiler from the build files.

[RFC=0002](https://github.com/googlecartographer/rfcs/blob/master/text/0002-cloud-based-mapping-1.md)

### CODE REVIEW

File: cartographer_grpc/mapping/trajectory_builder_stub.h [hunk 1]
```diff
@@ -57,27 +63,26 @@ class TrajectoryBuilderStub
...
 
  private:
-  struct LocalSlamResultReader {
-    grpc::ClientContext client_context;
-    std::unique_ptr<grpc::ClientReader<proto::ReceiveLocalSlamResultsResponse>>
-        client_reader;
-    std::unique_ptr<std::thread> thread;
-  };
-
   static void RunLocalSlamResultReader(
```
Comment (gaschler) [inline cartographer_grpc/mapping/trajectory_builder_stub.h]: naming consistency with thread and handler: RunLocalSlamResult*s*Reader?

File: cartographer_grpc/local_trajectory_uploader.cc [hunk 2]
```diff
@@ -31,22 +32,25 @@ const cartographer::common::Duration kPopTimeout =
...
+    CHECK(add_imu_client_->Finish().ok());
+  }
+  if (add_odometry_client_) {
+    CHECK(add_odometry_client_->WritesDone());
+    CHECK(add_odometry_client_->Finish().ok());
   }
-  if (odometry_writer_.client_writer) {
-    CHECK(odometry_writer_.client_writer->WritesDone());
-    CHECK(odometry_writer_.client_writer->Finish().ok());
+  if (add_fixed_frame_pose_client_) {
```
Comment (gaschler) [inline cartographer_grpc/local_trajectory_uploader.cc]: add_landmark_client_? (or add a todo and add it separately)

File: cartographer_grpc/mapping/pose_graph_stub.cc [hunk 3]
```diff
@@ -44,15 +48,14 @@ cartographer::mapping::MapById<
     cartographer::mapping::SubmapId,
     cartographer::mapping::PoseGraphInterface::SubmapPose>
 PoseGraphStub::GetAllSubmapPoses() {
-  grpc::ClientContext client_context;
   google::protobuf::Empty request;
-  proto::GetAllSubmapPosesResponse response;
-  stub_->GetAllSubmapPoses(&client_context, request, &response);
+  framework::Client<handlers::GetAllSubmapPosesHandler> client(client_channel_);
+  CHECK(client.Write(request));
```
Comment (gaschler) [inline cartographer_grpc/mapping/pose_graph_stub.cc]: (this is blocking for non-streaming handlers, right?)

File: cartographer_grpc/local_trajectory_uploader.cc (hunk 2 above)
Comment (cschuet) [reply cartographer_grpc/local_trajectory_uploader.cc]: done.

File: cartographer_grpc/mapping/pose_graph_stub.cc (hunk 3 above)
Comment (cschuet) [reply cartographer_grpc/mapping/pose_graph_stub.cc]: Correct.

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user cschuet. Merge queue now has a length of 1.

**Pull request merged**